src/Symfony/Bridge/Doctrine/Test/TestRepositoryFactory.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Test;
+
+use Doctrine\Common\Persistence\ObjectRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Repository\RepositoryFactory;
+
+/**
+ * @author Andreas Braun <alcaeus@alcaeus.org>
+ */
+final class TestRepositoryFactory implements RepositoryFactory
+{
+    /**
+     * The list of EntityRepository instances.
+     *
+     * @var \Doctrine\Common\Persistence\ObjectRepository[]
+     */
+    private $repositoryList = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRepository(EntityManagerInterface $entityManager, $entityName)
+    {
+        $repositoryHash = $this->getRepositoryHash($entityManager, $entityName);
+
+        if (isset($this->repositoryList[$repositoryHash])) {
+            return $this->repositoryList[$repositoryHash];
+        }
+
+        return $this->repositoryList[$repositoryHash] = $this->createRepository($entityManager, $entityName);
+    }
+
+    /**
+     * @param EntityManagerInterface $entityManager
+     * @param $entityName
+     * @param ObjectRepository $repository
+     */
+    public function setRepository(EntityManagerInterface $entityManager, $entityName, ObjectRepository $repository)
+    {
+        $repositoryHash = $this->getRepositoryHash($entityManager, $entityName);
+
+        $this->repositoryList[$repositoryHash] = $repository;
+    }
+
+    /**
+     * Create a new repository instance for an entity class.
+     *
+     * @param \Doctrine\ORM\EntityManagerInterface $entityManager The EntityManager instance.
```

Comment:
I'm not sure I understand correctly. Do you mean the method description, parameter description or both?